### ISSUE TITLE
[Feature Request] Checking Space around Bold and Italic

### ISSUE THREAD

**Issue by fyc09** [DavidAnson/markdownlint#1298]
Adding spaces around embedded code, bold, and italics is common practice to prevent unexpected rendering errors and make words recognizable. While Chinese can be recognized without spaces, adding spaces can enhance clarity and readability.

Correct:

```markdown
some *important* point
some `inline` code
一些 *重要* 的信息
一些 `inline` 代码
```

some *important* point
some `inline` code
一些 *重要* 的信息
一些 `inline` 代码

Incorrect:

```markdown
some*important*point
some`inline`code
一些*重要*的信息
一些`inline`代码
```

some*important*point
some`inline`code
一些*重要*的信息
一些`inline`代码

**Comment by fyc09** [comment]
@DavidAnson Can we use micromark as the parser in customRules to simplify column calculations?
https://github.com/DavidAnson/markdownlint/blob/b2305efafb034b1f328845aec9928b5363ffd646/lib/markdownlint.js#L58-L66

**Comment by DavidAnson** [comment]
Not yet. I want to convert the remaining core rules to micromark first, then review the API. Custom rules can still use markdown-it and the "helpers" package, but I am slowly removing things from that as I migrate, so feel free to use or copy from earlier versions.

**Comment by Faker095** [comment]
if ( 
   !result && 
   (rule.parser !== undefined) && 
   (rule.parser !== "markdownit") && 
   !((customIndex < 0) && (rule.parser === "micromark")) && 
   (rule.parser !== "none") 
 ) { 
   result = newError("parser", rule.parser); 
 }

**Comment by DavidAnson** [comment]
@fyc09, i'm not sure what the comment above is saying, but to get back to your question from six months ago, it is now possible to write custom rules that use the micromark parser. As you say, the line/column information should be easier to work with in that form. If you have any trouble, please let me know!
